### PR TITLE
new_design: Alternate proof formulation, maybe more readable?

### PR DESCRIPTION
**Pull request by eric-wieser**
Feel free to take bits piecewise

### CODE REVIEW

File: src/geometric_algebra/nursery/talk.lean [hunk 1]
```diff
@@ -86,49 +94,54 @@ variables (K : Type u) [field K]
 
 variables (V : Type u) [add_comm_group V] [vector_space K V]
 
-structure GA
-(G : Type u)
-[ring G] extends algebra K G :=
+#print linear_map
+
+def sqr {α : Type u} [has_mul α] (x : α) := x*x
+local postfix `²`:80 := sqr
```
Comment (eric-wieser) [inline src/geometric_algebra/nursery/talk.lean:100]: This avoids the nasty lambda functions in the expressions, and shows `²` in the tactic state for as long as possible.

File: src/geometric_algebra/nursery/talk.lean [hunk 2]
```diff
@@ -86,49 +94,54 @@ variables (K : Type u) [field K]
...
+
+structure GA (G : Type u) [ring G] [algebra K G] :=
 (fₛ : K →+* G)
-(fᵥ : V →+ G)
+(fᵥ : V →ₗ[K] G)
```
Comment (eric-wieser) [inline src/geometric_algebra/nursery/talk.lean:104]: I think this is more restrictive, and possibly more correct?

File: src/geometric_algebra/nursery/talk.lean [hunk 3]
```diff
@@ -86,49 +94,54 @@ variables (K : Type u) [field K]
 
 variables (V : Type u) [add_comm_group V] [vector_space K V]
 
-structure GA
-(G : Type u)
-[ring G] extends algebra K G :=
```
Comment (eric-wieser) [inline src/geometric_algebra/nursery/talk.lean:91]: Apparently `extends` is a terrible idea if the target has fewer arguments than the currently declared type

File: src/geometric_algebra/nursery/talk.lean [hunk 4]
```diff
@@ -86,49 +94,54 @@ variables (K : Type u) [field K]
...
-  obtain ⟨kbₛ, hb⟩ := GA.contraction ga (bᵥ),
-  have h2 : ga.fₛ (kabₛ - kaₛ - kbₛ) = a * b + b * a, by {
-    repeat {rw ring_hom.map_sub},
-    apply_fun (λ x, x - b * b - a * a) at h1,
-    simp [] at h1 ha hb hab,
-    simp [←h1, ha, hb, hab],
+  -- some tricks to unfold the `let`s and make things look tidy
+  delta,
+  set a := ga.fᵥ aᵥ,
+  set b := ga.fᵥ bᵥ,
```
Comment (eric-wieser) [inline src/geometric_algebra/nursery/talk.lean:123]: This seems like a super handy command to remember

**Review by utensil — APPROVED** [review]
Generally I like every bit!

**Pull request merged**